src/type-generation.ts

```diff
@@ -0,0 +1,233 @@
+type Type =
```

Comment:
I recommend not having a heart attack.

I can add tests later, but I'm mostly really happy that I managed to write this entire file in one sitting and it just _worked_.

Reply:
will review this file some other day :)